test/unit/error.spec.ts

```diff
@@ -292,3 +286,100 @@ test("return from nested finally", () => {
     `;
     expect(util.transpileAndExecute(code)).toBe("finally AB");
 });
+
+test("throw and catch custom error object", () => {
+    util.testFunction`
+      try {
+          throw { x: "Hello error object!" };
+      } catch (error) {
+          return error.x;
+      }
+    `.expectToMatchJsResult();
+});
+
+test.each(["Error", "RangeError", "ReferenceError", "SyntaxError", "TypeError", "URIError"])(
+    "throw builtin Errors as classes",
+    errorType => {
+        util.testFunction`
+            try {
+                throw new ${errorType}("message")
+            } catch (error) {
+                if (error instanceof Error) {
+                    return \`\${error}\`;
+                } else {
+                    throw Error("This error serves to prevent false positives from .expectNoExecutionError()");
+                }
+            }
+        `
+            .expectNoExecutionError()
+            .expectToMatchJsResult();
+    }
+);
+
+test.each(["Error", "RangeError", "ReferenceError", "SyntaxError", "TypeError", "URIError"])(
+    "throw builtin Errors as functions",
+    errorType => {
+        util.testFunction`
+            try {
+                throw ${errorType}("message")
+            } catch (error) {
+                if (error instanceof Error) {
+                    return \`\${error}\`;
+                } else {
+                    throw Error("This error serves to prevent false positives from .expectNoExecutionError()");
+                }
+            }
+        `
+            .expectNoExecutionError()
+            .expectToMatchJsResult();
+    }
+);
+
+test("subclass Error", () => {
+    util.testFunction`
+        class MyError extends Error {
+            name: "MyError"
+        }
+        
+        try {
+            throw new MyError();
+        } catch (error) {
+            if (error instanceof Error) {
+                return \'\$error\';
+            } else {
+                throw Error("This error serves to prevent false positives from .expectNoExecutionError()");
+            }
+        }
+    `
+        .expectNoExecutionError()
+        .expectToMatchJsResult();
+});
+
+test.each([["Error", false], ["Error", true], ["RangeError", false], ["MyError", false]] as Array<[string, boolean]>)(
+    "get stack from error",
+    (errorType, sourceMapTraceback) => {
+        const stack = util.testFunction`
+        class MyError extends Error {
+            name: "MyError"
```

Comment:
```suggestion
            public name = "MyError";
```

Reply:
Unresolved it since it's still an issue.
Test probably didn't fail because of `return \'\$error\';`

Reply:
Oh yeah I there was another one. Tests did't fail as we use `expectToMatchJsResult()` which doesn't care about the type definition.